modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/BlockListener.java

```diff
@@ -242,4 +242,18 @@ public void onFlow(BlockFromToEvent e){
             }
         }
     }
+
+    @EventHandler
+    public void onIceForm(BlockFormEvent e) {
+        if (e.isCancelled() || !Settings.DisableBlockForm) {
+            return;
+        }
+        MovecraftLocation loc = MathUtils.bukkit2MovecraftLoc(e.getBlock().getLocation());
+        for (Craft craft : CraftManager.getInstance().getCraftsInWorld(e.getBlock().getWorld())) {
+            if (craft.getHitBox().contains((loc))) {
+                e.setCancelled(true);
+                return;
+            }
```

Comment:
It would be faster to find nearest craft to location and then check if the block is within that hitbox.

Reply:
Oh, and the function name should change.

Reply:
Done